dao/FashionKnex.js

```diff
@@ -0,0 +1,75 @@
+const knex = require('knex')
+const Model = require('fashion-model/Model')
+const conflogger = require('conflogger')
+const _getTableName = require('./util/getTableName')
+
+function _clean (document) {
+  return Model.isModel(document)
+    ? Model.clean(document)
+    : document
+}
+
+module.exports = class FashionKnex {
+  constructor (options) {
+    let {
+      modelType,
+      tableName,
+      logger,
+      knexConfig
+    } = options
+
+    this._logger = conflogger.configure(logger)
+    this._modelType = modelType
+    this._tableName = tableName || _getTableName(modelType)
+    this._knex = knex(knexConfig)
+  }
+
+  async insert (data, returning) {
+    data = _clean(data)
```

Comment:
If we plan on enforcing a schema when the table is created than I'm cool with these changes. It's just that at the moment, it looks like any sort of data can be pushed into the table atm (postgres is schemaless so it would allow it).